### PR TITLE
Tool to run strace on a running VoltDB process

### PR DESCRIPTION
**Pull request by rmorgenstein**
I know I don't need a review to push a tool we don't ship, but I appreciate any python tips, ideas, corrections that anyone wants to provide.

This tools will find the VoltDB pid, or take a pid of any JVM-based process and run strace on its children. The output from strace replaces all the child pids with the thread names.

### CODE REVIEW

File: tools/stracevoltdb.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+except:
+    sys.exit ('ERROR: ' + args.regex + ' is not a valid regex')
+
+
+# Find the threads and make dict of pids.
+threads = find_threads(args.pid)
+
+
+#Find the ones that match the regex
+matching_threads = {key:value for key, value in threads.items() if re.search(pattern_thread, key)}
```
Comment (rmorgenstein) [inline tools/stracevoltdb.py]: Centos 6 has Python2.6, which doesn't have dictionary comprehensions. Rewrite

File: tools/stracevoltdb.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+if args.list:
+    exit()
+
+
+def replacepid(m):
+    return ('{0:5} - {1:25}'.format(m.group(1), pids[m.group(1)][:25]))
+
+#TODO: This could use better error handling for ctrl-c
+for line in run_cmd(strace_cmd, stderr_to_stdout=True):
```
Comment (rmorgenstein) [inline tools/stracevoltdb.py]: The output leaves a lot to be desired when strace isn't installed or can't be run without sudo.

File: tools/stracevoltdb.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    popen.stdout.close()
+    return_code = popen.wait()
+    if return_code != 0:
+        raise subprocess.CalledProcessError(return_code, cmd)
+
+
+desc = """Run strace on VoltDB (or another JVM-based process) and show output with
+thread names for all the activity.  Threads can be narrowed down by a regex.
+    Example - strace all threads with "Snap" in the name, but exclude futex calls
+    /stracethread.py -s "-e trace=\!futex" Snap.\*
```
Comment (pdshaw) [inline tools/stracevoltdb.py]: ./stracethread.py

File: tools/stracevoltdb.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    print
+    print "Strace command:"
+    print strace_cmd
+
+if args.list:
+    exit()
+
+
+def replacepid(m):
+    return ('{0:5} - {1:25}'.format(m.group(1), pids[m.group(1)][:25]))
```
Comment (pdshaw) [inline tools/stracevoltdb.py]: I prefer not to intermix methods and "main" code. Suggest you move this up and add a explicit "main":
   #### Command line main
   if __name__ == '__main__':
   ... the main stuff

**Pull request closed without merging**